R/deploy-site.R

```diff
@@ -0,0 +1,89 @@
+#' Deploy a pkgdown site on Travis-CI using Github Pages
+#'
+#' `deploy_site_github()` sets up your SSH keys for deployment, builds the
+#' site with [build_site()], commits the site to the `gh-pages` branch and then pushes
+#' the results back to GitHub. `deploy_site_github()` is meant only to be used
+#' by the CI system on Travis, it should not be called locally. See 'Setup' for
+#' details on setting up your repository to use this.
+#'
+#' @section Setup:
+#' Add the following to your `.travis.yml` file.
+#'
+#' ```
+#' r_github_packages: r-lib/pkgdown
```

Comment:
Or is it better to put `pkgdown` in suggests? I guess it has a lot of dependencies and if you have a build matrix you only need to install it on r-release? OTOH shouldn't we be consistent with covr?

Reply:
It would be nice to be consistent; but pkgdown has a _lot_ of dependencies.

We also probably only want to build this on the master branch and not for pull requests? So maybe this should be expanded to use travis jobs or a build matrix instead.